test/e2e/windows/device_plugin.go

```diff
@@ -0,0 +1,150 @@
+
+	"github.com/onsi/ginkgo"
+)
+
+var _ = SIGDescribe("Device Plugin", func() {
+	f := framework.NewDefaultFramework("device-plugin")
+
+	var cs clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		//Only for Windows containers
+		e2eskipper.SkipUnlessNodeOSDistroIs("windows")
+		cs = f.ClientSet
+	})
+	ginkgo.It("should be able to create a functioning device plugin for Windows", func() {
+		ginkgo.By("creating Windows device plugin daemonset")
+		dsName := "directx-device-plugin"
+		daemonsetNameLabel := "daemonset-name"
+		image := "e2eteam/k8s-directx-device-plugin:0.9.0-1809"
+		mountName := "device-plugin"
+		mountPath := "/var/lib/kubelet/device-plugins"
+		privileged := true
+		labels := map[string]string{
+			daemonsetNameLabel: dsName,
+		}
+        	ds := &appsv1.DaemonSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: dsName,
+				Namespace: "kube-system",
+			},
+			Spec: appsv1.DaemonSetSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: labels,
+				},
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"scheduler.alpha.kubernetes.io/critical-pod": "",
+						},
+						Labels: labels,
+					},
+					Spec: v1.PodSpec{
+						Tolerations: []v1.Toleration{
+							{
+								Key: "CriticalAddonsOnly",
+								Operator: "Exists",
+							},
+						},
+						Containers: []v1.Container{
+							{
+								Name:  "hostdev",
+								Image: image,
+								SecurityContext: &v1.SecurityContext{
+									Privileged: &privileged,
+								},
+								VolumeMounts: []v1.VolumeMount{
+									{
+										Name: mountName,
+										MountPath: mountPath,
+									},
+								},
+								Env: []v1.EnvVar{
+									{
+										Name: "DIRECTX_GPU_MATCH_NAME",
+										Value: " ",
+									},
+								},
+							},
+						},
+						Volumes: []v1.Volume{
+							{
+								Name: mountName,
+								VolumeSource: v1.VolumeSource{
+									HostPath: &v1.HostPathVolumeSource{
+										Path: mountPath,
+									},
+								},
+							},
+						},
+						NodeSelector: map[string]string{
+							"kubernetes.io/os": "windows",
+						},
+					},
+				},
+			},
+		}
+
+		sysNs := "kube-system"
+		ds, err := cs.AppsV1().DaemonSets(sysNs).Create(context.TODO(), ds, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("creating Windows testing Pod")
+		windowsPod := createTestPod(f, imageutils.GetE2EImage(imageutils.WindowsServer), windowsOS)
+		windowsPod.Spec.Containers[0].Args = []string{"powershell.exe", "Start-Sleep", "3600" }
+		windowsPod.Spec.Containers[0].Resources.Limits = v1.ResourceList{
+			"microsoft.com/directx": resource.MustParse("1"),
+		}
+		windowsPod = f.PodClient().CreateSync(windowsPod)
+
+		ginkgo.By("verifying device access in Windows testing Pod")
+		dxdiagCommand := []string{"cmd.exe", "/c", "dxdiag", "/t", "dxdiag_output.txt", "&", "type", "dxdiag_output.txt"}
+		dxdiagDirectxVersion := "DirectX Version: DirectX 12"
```

Comment:
Should the hard-coded version checks for DirectX and DirectX DDI require that the OS-selectors specified above in the pod yaml specifically look for Windows OS versions that are expected to have DirectX 12 and DDI 12? Otherwise these tests would fail. Maybe for now, add a comment here that down the line, when DirectX version bumps occur in a subsequent release of Windows, these tests will need to do version checks depending on the Windows version running the test.